org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/debug/FqnDiscover.java

```diff
@@ -178,30 +174,32 @@ public static String identifyFqnInResource(List<Object> params, IProgressMonitor
    * @return all resources are identified by the given fqn
    */
   public static List<Either<String, ResourceLocation>> findResourcesByFqn(
-      List<Object> params, IProgressMonitor pm) {
+      List<Object> params, IProgressMonitor progressMonitor) {
     Preconditions.checkArgument(params.size() >= 1, "FQN is expected.");
 
+    ensureNotCancelled(progressMonitor);
+
     String fqn = (String) params.get(0);
+    List<IJavaProject> projects = getWorkspaceJavaProjects();
```

Comment:
Unfortunately, we don't control the order or lookup here, but this is a deeper problem that is not part of this PR.